spanner/spanner_snippets/spanner/spanner_create_instance_with_autoscaling_config.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_create_instance_with_autoscaling_config]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"cloud.google.com/go/spanner/admin/instance/apiv1/instancepb"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+func createInstanceWithAutoscalingConfig(w io.Writer, projectID, instanceID string) error {
+	// projectID := "my-project-id"
+	// instanceID := "my-instance"
+	ctx := context.Background()
+	instanceAdmin, err := instance.NewInstanceAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("could not create instance admin client for project %s: %w", projectID, err)
+	}
+	defer instanceAdmin.Close()
+
+	instanceName := fmt.Sprintf("projects/%s/instances/%s", projectID, instanceID)
+	fmt.Fprintf(w, "Creating instance %s.", instanceName)
+
+	op, err := instanceAdmin.CreateInstance(ctx, &instancepb.CreateInstanceRequest{
+		Parent:     fmt.Sprintf("projects/%s", projectID),
+		InstanceId: instanceID,
+		Instance: &instancepb.Instance{
+			Config:      fmt.Sprintf("projects/%s/instanceConfigs/%s", projectID, "regional-us-central1"),
+			DisplayName: "This is a display name.",
```

Comment:
nit: use a more descriptive display name. Customers may run this snippet and forget about it.

Suggestion:

```
DisplayName: "Example of creating an instance programmatically",
```

Reply:
Updated. PTAL.